src/Parser/MarksmanshipHunter/Modules/FocusChart/Focus.js

```diff
@@ -0,0 +1,353 @@
+//Based on Main/Mana.js and Parser/VengeanceDemonHunter/Modules/PainChart
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import ChartistGraph from 'react-chartist';
+import Chartist from 'chartist';
+import 'chartist-plugin-legend';
+
+import makeWclUrl from 'common/makeWclUrl';
+import SPELLS from 'common/SPELLS';
+
+import specialEventIndicators from 'Main/Chartist/specialEventIndicators';
+
+import { formatDuration } from 'common/format';
+
+import 'Main/Mana.css';
+
+import FocusComponent from './FocusComponent';
+import './Focus.css';
+
+
+const passiveWasteThresholdPercentage = .03; // (wasted passive focus generated) / (total passive focus generated), anything higher will trigger "CAN BE IMPROVED"
+//TODO: get a "real" number approved by a MMS expert
+
+class Focus extends React.PureComponent {
+  static propTypes = {
+    reportCode: PropTypes.string.isRequired,
+    actorId: PropTypes.number.isRequired,
+    start: PropTypes.number.isRequired,
+    end: PropTypes.number.isRequired,
+    playerHaste: PropTypes.number.isRequired,
+    focusMax: PropTypes.number,
+    passiveWaste: PropTypes.array,
+    pWTracker : PropTypes.number,
+    activeWaste: PropTypes.array,
+    aWTracker: PropTypes.number,
+  };
+
+  constructor() {
+    super();
+
+    this.state = {
+      focusData: null,
+      bossHealth: null,
+    };
+	
+  }
+
+  componentWillMount() {
+    this.load(this.props.reportCode, this.props.actorId, this.props.start, this.props.end);      
+    }
+
+  componentWillReceiveProps(newProps) {
+    if (newProps.reportCode !== this.props.reportCode || this.props.focusMax !== this.props.focusMax || this.props.pWTracker !== this.props.pWTracker || this.props.aWTracker !== this.props.aWTracker || newProps.actorId !== this.props.actorId || newProps.start !== this.props.start || newProps.end !== this.props.end) {
+      this.load(newProps.reportCode, newProps.actorId, newProps.start, newProps.end);
+    }
+  }
+  load(reportCode, actorId, start, end) {
+
+ 
+    const focusPromise = fetch(makeWclUrl(`report/tables/resources/${reportCode}`, {
```

Comment:
This should already be available in the events under the classResources property. Caching this in an array should provide you with the same information without the need for an additional API call.

The events we fetch are a complete set with all data that's available from the combatlog (and thus also the only thing that WCL uses). The only thing we don't receive are events not directly related to the selected player. Any additional API calls for information about just the selected player should therefore be unnecessary. 

In this case I noticed you're also using the deaths property of the response, but that should be the same on the bossHealthPromise response and should be able to be used instead.